### ISSUE TITLE
Uncensor

### ISSUE THREAD

**Issue by chaos4455** [bes-dev/stable_diffusion.openvino#53]
the project is very good but it look has it have some censorship about nudity. Are possible to turn off al the filters from the models used?

**Comment by Mikhael-Danilov** [comment]
I believe this implementation don't have any nudity filters built-in.

**Comment by bes-dev** [comment]
@chaos4455 we don't use any safety checkers for output images, but we hope that users will not use our project to create inappropriate or illegal content.

**Comment by chaos4455** [comment]
@bes-dev , thanks for reply. "but we hope that users will not use our project to create inappropriate or illegal content." I hope it too, unfortunately from my vision we have no control on that. You make a very grood work, very good. 

Here an example. I use some input with sexy and sensual word as nude and i dont give any nude result.

Examples: Looks like the model try to censor nude parts or words relative to those themes.



![output](https://user-images.githubusercontent.com/30844283/189212278-7017e035-969c-4bc8-9bf7-4a43d5d64151.png)
![fgdgdf](https://user-images.githubusercontent.com/30844283/189212285-3f2bb3e5-aa49-4fdd-86ad-1708a07bc681.png)
![gfdgfd](https://user-images.githubusercontent.com/30844283/189212288-79c17030-7eae-40ad-9306-6ff9554e43e7.png)
![fdfdf](https://user-images.githubusercontent.com/30844283/189212291-3dc2aca2-81f3-4a74-9556-867b3890d3ea.png)


Otherwise, if we put something random the results look preety awesome.

![sdf](https://user-images.githubusercontent.com/30844283/189212427-df4d8894-a5b4-4672-8e96-9d9e1d522046.png)
![aaa](https://user-images.githubusercontent.com/30844283/189212433-6fa24d25-3a3e-4f81-a56a-67e4aec138fb.png)
![ddd](https://user-images.githubusercontent.com/30844283/189212436-83e148b8-d869-4524-8172-b19d963b2471.png)
![aaaa](https://user-images.githubusercontent.com/30844283/189212438-879e2f0c-0aa8-4e1f-aa3b-8dec1c09da0f.png)

**Comment by chaos4455** [comment]
![fdf](https://user-images.githubusercontent.com/30844283/189213612-77776526-929e-445f-b642-dda3d82f0682.png)
Here an example, the model try to hide the sexual body parts. Thanks again!

**Comment by Mikhael-Danilov** [comment]
> ![fdf](https://user-images.githubusercontent.com/30844283/189213612-77776526-929e-445f-b642-dda3d82f0682.png) Here an example, the model try to hide the sexual body parts. Thanks again!

This is because not a lot of explicit material included in [laion](laion.ai) dataset (on which SD was trained) (and probably it is additionally filtered before training SD). So SD as is not suitable as porn generator. (Architecture itself is pretty capable to do it, yet proper dataset and a lot of computing power required to achieve this goal)

**Comment by chaos4455** [comment]
Thanks for the answer, is possible to edit the code of this project to use a more suitable dataset? i dont have fhurther klowledgement about this but if you tell me some good information i think im capable to do that. Are possible to use others dataset in this projetc? Can you please help me to do that? i see that when i start the project fisrt time it download a 3gb dataset. Are possible do force the project to use other more suitable to this?

**Comment by Zoranvedek** [comment]
There is no censorship happening here, I have folders full images containing artistic nudes, both intentional and accidental.  I can easily assure you that there is nothing being censored:

![image](https://user-images.githubusercontent.com/112593851/189238441-dbc7f3d7-b324-49c4-8f70-65c3d392d930.png)
![image](https://user-images.githubusercontent.com/112593851/189238481-771582cb-3d59-4e45-aa14-4505f9d035b4.png)
![image](https://user-images.githubusercontent.com/112593851/189238513-49dc7113-0c4c-4346-946f-cdec72224687.png)


....now with that said, this is not a random porn generation tool, and I'd hate to see problems arising from it's use as such. Honestly what you're likely seeing is a limitation of the training set of images, not intentional censorship.

**Comment by fhaust** [comment]
The original Laion5B dataset does have a "NSFW" score:

>  Additionally, we provide several nearest neighbor indices, an improved web interface for exploration & subset creation as well as detection scores for watermark and NSFW.

I do not have a reference for that, but I think the stable diffusion was trained on a subset of Laion5B with the NSFW tagged content removed (or better: not included). As such the model has "seen" less nudity and explicit content than it could have seen.

That said, it certainly has seen nudity and can be coerced into reproducing it with (un-)appropriate prompts.

@chaos4455 The thing is that it is not as simple as using "a more suitable dataset". There literally is no (open) alternative to the [Stable Diffusion](https://github.com/CompVis/stable-diffusion) model that this project relies on. And (if I am not mistaken) that cost USD 600000 to train when it was first released.

(Sidenote: am I the only one not trying to produce porn with stable diffusion? :-) )

**Comment by chaos4455** [comment]
Yeah, thanks, thats right: For the moment this is preety impressive. Im using ain i39100f and the performance is quite impressive too. arround 6,3 sec for instance. Thanks for the information you give to us.

**Comment by Mikhael-Danilov** [comment]
However, fine tuning should be possible, and not as expensive as initial training. And we already have proof of it: https://github.com/harubaru/waifu-diffusion
